src/cfn-resource-generator.ts

```diff
@@ -0,0 +1,162 @@
+import * as camelcase from 'camelcase';
+import * as j2j from 'json2jsii';
+import { sanitizeTypeName } from './util';
+import { TypeInfo } from './type-info';
+
+/**
+ * Generator to emit classes and types for the L1 construct of the given resource type
+ */
+export class CfnResourceGenerator {
+
+  sanitizedTypeName: string;
+  readOnlyProperties: string[];
+  writeProperties: string[];
+  constructClassName: string;
+  propsClassName: string;
+
+  /**
+   * 
+   * @param typeName the name of the CFN resource type (e.g. AWSQS::EKS::Cluster)
+   * @param typeDef the type info containing the source url
+   * @param schema the schema of the resource type for input and output properties
+   */
+  constructor(private readonly typeName: string, private readonly typeDef: TypeInfo, private readonly schema: any) {
+    this.sanitizedTypeName = sanitizeTypeName(typeName);
```

Comment:
```suggestion
    this.typeName = sanitizeTypeName(typeName);
```

Reply:
this.typeName is the name with `::` and needs to remain. Why do you want to overwrite this value?